src/include/stir/IO/SAFIRCListmodeInputFileFormat.h

```diff
@@ -90,16 +90,17 @@ class SAFIRCListmodeInputFileFormat : public InputFileFormat<CListModeData>, pub
 		std::ifstream par_file(filename.c_str());
 		std::string key;
 		std::getline(par_file, key, ':');
-		key.erase(std::remove_if(key.begin(), key.end(), is_whitespace), key.end());
-		if( !boost::iequals(key, std::string("CListModeDataSAFIRParameters")) ) { 
+		key = standardise_interfile_keyword(key);
+		if( key != std::string("clistmodedatasafir parameters")) { 
 			return false;
 		}
 
 		bool can_parse = actual_do_parsing(filename);
+		if( !can_parse ) return false;
 		std::ifstream data_file(listmode_filename.c_str(), std::ios::binary);
 		char* buffer = new char[32];
 		data_file.read(buffer, 32);
-		bool cr = (!strncmp(buffer, "MUPET CListModeData\0", 20) ||  !strncmp(buffer, "SAFIR CListModeData\0", 20)) && can_parse;
+		bool cr = (!strncmp(buffer, "MUPET CListModeData\0", 20) ||  !strncmp(buffer, "SAFIR CListModeData\0", 20));
```

Comment:
seems to me that this could use writing 
```
warning(boost::format("SAFIR tried to read file '%1%' but this seems to be in the wrong format") % listmode_filename);
```
otherwise it will be a silent warning and people will get very puzzled.

if you think this is worth adding, please test it on a file first (as I've had that the warning itself crashed because I mistyped the format string).